### ISSUE TITLE
Prevent associated record modal dialog from showing when associated record requires an attachment

### ISSUE THREAD

**Issue by murdoch** [railsadminteam/rails_admin#663]
I love the way rails admin allows you to create associated records through a modal dialog. However, if the associated record being created has an image attachment then the modal form is not submitted via ajax. Instead it gets processed via http. This means that there is a redirect in the parent view whenever an associated record that requires a file attachment is submitted through the modal dialog. All data in the parent view is lost and the user is redirected to the list view for the associated record.

Not ideal.

So I'm wondering if it's possible to check in advance, whether or not an associated record requires an attachment, like a pdf or a jpg, and if it does, then possibly just hide the link that fires up the dialog box. It's not an ideal solution but it does prevent frustrating data loss.

At the moment, I am actually having to supress all of these dialogs by commenting out some code in views/rails_admin/main/_form_filtering_multiselect.html.haml because so many of my models have attachments. Which is a shame because I do like the modal dialogs

```
- if authorized? :new, config.abstract_model
  - path_hash = { :model_name => config.abstract_model.to_param }
  - path_hash.merge!({ :associations => { field.inverse_of => (@object.id || 'new') } }) if field.inverse_of
  // I'm preventing all modal dialogs from being accessed because so many of my models have attachements
  // = link_to t('admin.actions.create').capitalize + ' ' + config.abstract_model.pretty_name.downcase, rails_admin_new_path(path_hash), :class => "createAssociatedRecord"
```
